### DISCUSSION TITLE
`get_unique_constraints` support for mssql dialect?

### DISCUSSION THREAD

**Discussion by peanutcolada** [sqlalchemy/sqlalchemy#8317] [Usage Questions] [answered]
I am trying to use the inspector to get the unique constraints for an existing table on MS SQL using this https://docs.sqlalchemy.org/en/14/core/reflection.html#sqlalchemy.engine.reflection.Inspector.get_unique_constraints

But oddly enough there are no documentations about how `get_unique_constraints` is not implemented with the mssql dialect or am I missing something? Is there a custom implementation somewhere I can use temporarily?

**Comment by zzzeek** [comment, marked as answer]
the MSSQL dialect doesn't have a unique constraints method because these are returned as indexes with the unique flag set , so I would use get_indexes() instead.

**Reply by peanutcolada** [reply]
ah got it, is this the same situation with `get_check_constraints` as well?

**Reply by zzzeek** [reply]
not sure about CHECK.  if they aren't there we might have them not implemented for SQL Server though that seems surprising

**Reply by CaselIT** [reply]
Yes mssql doesn't have check constraint reflection
